docs/src/main/asciidoc/kotlin.adoc

```diff
@@ -173,6 +173,203 @@ for example, then `<option>all-open:annotation=javax.enterprise.context.Applicat
+}
+----
+
+<1> The Quarkus plugin needs to be applied.
+<2> The Kotlin plugin version needs to be specified.
+<3> We include the Quarkus BOM using Gradle's link:https://docs.gradle.org/5.4.1/userguide/managing_transitive_dependencies.html#sec:bom_import[relevant syntax]
+<4> The rest-assured version needs to be specified
+<5> All open configuration required, as per Maven guide above
+
+or, if you use the Gradle Kotlin DSL:
+
+[source,kotlin,subs=attributes+]
+----
+import io.quarkus.gradle.tasks.QuarkusNative
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = "..."
+version = "1.0.0-SNAPSHOT"
+
+plugins {
+    java
+    id("io.quarkus") version "{quarkus-version}" // <1>
+
+    kotlin("jvm") version "{kotlin-version}" // <2>
+    id("org.jetbrains.kotlin.plugin.allopen") version "{kotlin-version}" // <2>
+}
+
+apply(plugin = "io.quarkus")
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation(kotlin("stdlib-jdk8"))
+
+    // <3>
+    implementation(enforcedPlatform("io.quarkus:quarkus-bom:${quarkusVersion}"))
+    implementation("io.quarkus:quarkus-kotlin")
+    implementation("io.quarkus:quarkus-resteasy")
+    implementation("io.quarkus:quarkus-resteasy-jsonb")
+
+    testImplementation("io.quarkus:quarkus-junit5")
+    testImplementation("io.rest-assured:rest-assured:{rest-assured-version}") // <4>
+}
+
+tasks {
+    named<QuarkusNative>("buildNative") {
+        setEnableHttpUrlHandler(true)
+    }
+}
+
+tasks.test {
+    useJUnitPlatform()
+    exclude '**/Native*'
+}
+
+allOpen { // <5>
+    annotation("javax.ws.rs.Path")
+    annotation("javax.enterprise.context.ApplicationScoped")
+    annotation("io.quarkus.test.junit.QuarkusTest")
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+val compileKotlin: KotlinCompile by tasks
+compileKotlin.kotlinOptions {
+    jvmTarget = "1.8"
+}
+
+val compileTestKotlin: KotlinCompile by tasks
+compileTestKotlin.kotlinOptions {
+    jvmTarget = "1.8"
+}
+----
+
+<1> The Quarkus plugin needs to be applied.
+<2> The Kotlin plugin version needs to be specified.
+<3> We include the Quarkus BOM using Gradle's link:https://docs.gradle.org/5.4.1/userguide/managing_transitive_dependencies.html#sec:bom_import[relevant syntax]
+<4> The rest-assured version needs to be specified
+<5> All open configuration required, as per Maven guide above
+
+== CDI @Inject with Kotlin
+
+Kotlin reflection annotation processing differs from Java.  You may experience an error when using CDI @Inject such as:
+"kotlin.UninitializedPropertyAccessException: lateinit property xxx has not been initialized"
+
+In the example below, this can be easily solved by adapting the annotation, adding @field: Default, to handle the lack of a @Target on the Kotlin reflection annotation definition. 
+
+Alternatively, prefer the use of the compile-time injection with ArC which works without modification of the Java examples.
```

Comment:
I think you mean `constructor injection` instead of `compile-time injection`, correct?

Reply:
Updated it to constructor injection.